game/plugin/skills/fishing/src/org/apollo/game/plugin/skills/fishing/FishingAction.kt

```diff
@@ -0,0 +1,88 @@
+package org.apollo.game.plugin.skills.fishing
+
+import org.apollo.game.action.ActionBlock
+import org.apollo.game.action.AsyncDistancedAction
+import org.apollo.game.model.entity.Player
+import org.apollo.game.plugin.api.fishing
+import java.util.Objects
+
+class FishingAction(
+    player: Player,
+    private val target: FishingTarget
+) : AsyncDistancedAction<Player>(0, true, player, target.position, SPOT_DISTANCE) {
+
+    /**
+     * The [FishingTool] used for the fishing spot.
+     */
+    private val tool = target.option.tool
+
+    override fun action(): ActionBlock = {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. ([kotlin:S3776](https://sonarcloud.io/organizations/github-apollo-rsps/rules#rule_key=kotlin%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=apollo%3Aorg.apollo&pullRequest=338&issues=AWVzgXw2XKENQn1Ssjbk&open=AWVzgXw2XKENQn1Ssjbk)

[ISSUE_KEY]: AWVzgXw2XKENQn1Ssjbk